packages/framer-motion/src/animation/js/index.ts

```diff
@@ -61,6 +61,22 @@ export function animateValue<V = number>({
     onUpdate,
     ...options
 }: AnimationOptions<V>): MainThreadAnimationControls<V> {
+    let resolveFinishedPromise: VoidFunction
+    let currentFinishedPromise: Promise<void>
+
+    /**
+     * Create a new finished Promise every time we entered the
+     * finished state and resolve the old Promise. This is
+     * WAAPI-compatible behaviour.
+     */
+    const updateFinishedPromise = () => {
+        currentFinishedPromise = new Promise((resolve) => {
+            resolveFinishedPromise = resolve
+        })
+    }
+
+    updateFinishedPromise()
```

Comment:
I guess this saves a few bytes because you don't have to initialise `currentFinishedPromise` with the same code in `updateFinishedPromise` but it confused me for a sec.

Maybe just a lil comment above saying `// Initialize the finished promise`